### DISCUSSION TITLE
upgrade from 2.1.1 to 2.2+: duplicate m2 serial

### DISCUSSION THREAD

**Discussion by Robby-Swartenbroekx** [WillCodeForCats/solaredge-modbus-multi#96] [Q&A] [answered]
I followed the procedure and first removed the integration from the settings page.
Then I did the upgrade.
Rebooted
Then I reinstalled the integration in the settings page, but I get the error: "Duplicate m2 serial 606857732"
even after rebooting again, I keep having this issue.

Glad I did make a esxi snapshot, reverting to that solved the issue.
I tried upgrading from 2.1.1 to 2.1.3 after that, but same issue.

Also removed the integration from HACS, reboot, and readd everything.

I'm guessing the "Duplicate m2 serial ..." message is actually an error that is returning over modbus, so something that the inverter is saying. Anybody know how I continue now? it works for me on 2.1.1, but I would like to keep following all updates, so if there is a solution, I would like to try it out.

**Comment by WillCodeForCats** [comment, marked as answer]
Please try using [Release v2.2.2](https://github.com/WillCodeForCats/solaredge-modbus-multi/releases/tag/v2.2.2) which should fix this issue.

**Reply by WillCodeForCats** [reply]
PR #81 created a regression with duplicate serial number detection originally in PR #51 which should be fixed with PR #98.

**Reply by WillCodeForCats** [reply]
Didn't get any feedback on the pre-release, but I didn't want to keep waiting, so I just made it into a full release with PR #98 and PR #94.
